torch/testing/asserts.py

```diff
@@ -0,0 +1,389 @@
+        b (torch.Tensor): Second tensor.
+        warn_floating_point_dtype (bool): If ``True`` (default), emit a warning if :attr:`a` or :attr:`b` is a
+            floating point tensor.
+        check_device (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` live in the same
+            :attr:`~torch.Tensor.device` memory. If this check is disabled **and** :attr:`a` and :attr:`b` do not live
+            in the same memory :attr:`~torch.Tensor.device`, they are moved to the CPU memory.
+        check_dtype (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` have the same
+            :attr:`~torch.Tensor.dtype`. If this check is disabled **and** :attr:`a` and :attr:`b` do not have the same
+            :attr:`~torch.Tensor.dtype`, the comparison :attr:`~torch.Tensor.dtype` is determined by
+            :func:`torch.promote_types`.
+        check_stride (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` have the same
+            :meth:`~torch.Tensor.stride`.
+
+    Raises:
+        AssertionError: If :attr:`a` and :attr:`b` do not have the same :attr:`~torch.Tensor.shape`.
+        AssertionError: If :attr:`check_device`, but :attr:`a` and :attr:`b` do not live in the same
+            :attr:`~torch.Tensor.device` memory.
+        AssertionError: If :attr:`check_dtype`, but :attr:`a` and :attr:`b` do not have the same
+            :attr:`~torch.Tensor.dtype`.
+        AssertionError: If :attr:`check_stride`, but :attr:`a` and :attr:`b` do not have the same
+            :meth:`~torch.Tensor.stride`.
+        AssertionError: If the values of :attr:`a` and :attr:`b` are not bitwise equal.
+        UsageError: If :attr:`a` or :attr:`b` is complex, quantized, or sparse. This is a temporary restriction and
+            will be relaxed in the future.
+
+    .. seealso::
+
+        To compare tensors for value proximity, :func:`assert_tensors_allclose` can be used.
+    """
+    _assert_are_tensors(a, b)
+
+    is_floating_point = a.dtype.is_floating_point or b.dtype.is_floating_point
+    if is_floating_point and warn_floating_point_dtype:
+        warnings.warn(
+            "If one tensor has an floating-point dtype, "
+            "it is recommended to use the respective 'allclose' variant instead.",
+            UsageWarning,
+        )
+
+    _compare_tensors(
+        a,
+        b,
+        values_asserter=_assert_values_equal,
+        check_device=check_device,
+        check_dtype=check_dtype,
+        check_stride=check_stride,
+    )
+
+
+@_hide_internal_traceback_pytest
+def assert_tensors_allclose(
+    a: torch.Tensor,
+    b: torch.Tensor,
+    /,
+    *,
+    warn_integer_dtypes: bool = True,
+    rtol: Optional[float] = None,
+    atol: Optional[float] = None,
+    check_device: bool = True,
+    check_dtype: bool = True,
+    check_stride: bool = True,
+) -> None:
+    """Asserts that the values of two tensors are equal up to a desired precision.
```

Comment:
"equal up to a desired precision" might be unclear to some users because it's suggestive of a pattern like, "assert these are equivalent up to this decimal place," which is possible to implement with this function but not a requirement.

It might be simpler to say "... of two tensors are close." for now